src/handformer.cpp

```diff
@@ -0,0 +1,64 @@
+#include "handformer.h"
+
+#include <algorithm>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "piecetype.h"
+
+using Mahjong::Piece;
+
+std::map<char, Piece::Type> NOTATION_TO_SUIT = {
+  {'m', Piece::CHARACTER_SUIT},
+  {'s', Piece::BAMBOO_SUIT},
+  {'p', Piece::PIN_SUIT},
+  {'z', Piece::HONOR_SUIT},
+};
+
+std::map<uint8_t, Piece::Type> NOTATION_TO_HONOR = {
+  {1, Piece::EAST_WIND},
+  {2, Piece::SOUTH_WIND},
+  {3, Piece::WEST_WIND},
+  {4, Piece::NORTH_WIND},
+  {5, Piece::WHITE_DRAGON},
+  {6, Piece::GREEN_DRAGON},
+  {7, Piece::RED_DRAGON},
+};
+
+auto Mahjong::HandFromNotation(std::string notation) -> std::vector<Piece> {
+  std::vector<Piece> result;
+  std::vector<int8_t> currentTiles;
+  for (const auto& c : notation) {
+    if (std::isdigit(c)) {
+      auto value = c - '0';
```

Comment:
This probably fine this stackoverflow answer points to this being safe with UTF-7 and UTF-8 and I don't think we need to support any additional encodings here. Consider a quick comment tho.